annotations/management/commands/import_tenses.py

```diff
@@ -30,59 +30,78 @@ def handle(self, *args, **options):
 
 
 def process_file(f, language, use_other_label, model='annotation'):
-    for n, row in enumerate(f):
-        if n == 0:
-            continue
-
-        row = row.strip()
+    f = iter(f)
+    header = next(f)
+    if isinstance(header, bytes):
+        header = header.decode()
+    header = header.strip().split('\t')
+
+    columns = []
+    for h in header[1:]:
+        if isinstance(h, bytes):
+            h = h.decode()
+        if h.lower() in ['tense', 'comments']:
+            columns.append(h)
+        else:
+            try:
+                key = LabelKey.objects.get(title__iexact=h)
+                columns.append(key)
+            except LabelKey.DoesNotExist:
+                raise ValueError('Unknown label "{}"'.format(h))
+
+    for row in f:
+        row = row.decode().strip()
         if row:
-            encoded = [c.decode('utf-8') for c in row.split('\t')]
+            encoded = row.split('\t')
 
             if model == 'annotation':
-                update_annotation(language, encoded, use_other_label)
+                obj = get_annotation(encoded)
             elif model == 'fragment':
-                update_fragment(language, encoded, use_other_label)
+                obj = get_fragment(language, encoded)
             else:
                 raise ValueError('Unknown model {}'.format(model))
 
+            update_fields(obj, language, encoded, columns)
 
-def update_annotation(language, row, use_other_label=False):
-    try:
-        # Retrieve Annotation
-        annotation = Annotation.objects.get(pk=row[0])
 
-        # Add Tense or other_label
-        if use_other_label:
-            annotation.other_label = row[1]
-        else:
-            annotation.tense = Tense.objects.get(title__iexact=row[1], language=language)
+def update_fields(obj, language, row, columns):
+    for idx, column in enumerate(columns):
+        if idx + 1 >= len(row):
+            continue
+        cell = row[idx + 1]
+        if column == 'tense':
+            try:
+                obj.tense = Tense.objects.get(title__iexact=cell, language=language)
+            except Tense.DoesNotExist:
+                raise ValueError('Tense with title "{}" not found.'.format(row[1]))
+        elif column == 'comments':
+            if isinstance(obj, Annotation):
+                obj.comments = cell
+            else:
+                raise ValueError('Cannot add comments to Fragment')
+        elif isinstance(column, LabelKey):
+            label, created = Label.objects.get_or_create(title=cell,
+                                                         key=column,
+                                                         language=language)
+            if created:
+                label.save()
 
-        # Add comments
-        if len(row) == 3:
-            annotation.comments = row[2]
+            obj.labels.add(label)
```

Comment:
What if the `obj` already has a label with this key? It would create two labels with the same key, right? That is not what we want, I think.

Reply:
as far as I can tell this doesn't cause duplication, so I'd rather leave it like that

Reply:
I'm sorry, I wasn't really clear in my first comment. Some steps to reproduce the issue:

1. I am working on Europarl-SINCE, I downloaded the Dutch Annotations to .xlsx, then changed all labels in the C-column to "sinds". 
2. I then uploaded the below .txt-file via the import functionality (I renamed the LabelSet to something more convenient :-))
```
id	tense	since-adverbial
62447	ott	sinds
62194	ott	sinds
```
3. Now I arrive at the following: 
```
$ python manage.py shell
>>> from annotations.models import Annotation
>>> a = Annotation.objects.get(pk=62194)
>>> a.get_labels()
'ott, al, sinds'
```
4. The problem now is that the last two labels have the same LabelKey, which is problematic I think. 

Do you agree?

Reply:
Thanks for making the problem clear! I missed it completely ;)